docs/search_space.md

```diff
@@ -54,10 +54,18 @@ means, a basic component of many HPO algorithms. The following domains are curre
 * `choice(categories)`: Uniform from the finite list `categories` of values.
   Entries in `categories` should ideally be of type `str`, but types `int` and
   `float` are also allowed (the latter can lead to errors due to round-off).
-* `ordinal(categories)`: Variant of `choice` for which the order of entries in
-  `categories` matters. Essentially, we use `randint(0, len(categories) - 1)`
-  internally on the positions in `categories`. For methods like Bayesian
-  optimization, nearby elements in the list have closer encodings.
+* `ordinal(categories, kind)`: Variant of `choice` for which the order of
+  entries in `categories` matters. For methods like Bayesian optimization,
+  nearby elements in the list have closer encodings. Compared to `choice`, the
+  encoding consists of a single number here. Different variants are implemented.
+  If `kind="equal"` (default), we use `randint(0, len(categories) - 1)`
+  internally on the positions in `categories`, so that each category is chosen
+  with the same probability. If `kind="nn"`, `categories` must contain strictly
```

Comment:
Careful. "equal" and "nn" are really different things. I'd also typically go for "nn" for numerical types, but maybe other users may not.

Indeed, the current setup does not enforce "best practices" well.

Reply:
I think the question whether you'd use "equal" or "nn" is not so easy to answer in general, no?
I'd intuitively rather go for "nn" or "nn-log", because that is close to just using a continuous relaxation.
But I cannot say this would always be better.

Of course, if you use "equal", the numbers do not make a difference, you could just as well call them 0, 1, 2,

Reply:
What I could do is test the values. If they are numerical and increasing, I could use "nn" as default instead of "equal".

We could even go as far and test whether they are equi-spaced, and if so, use findiff, but I find this gets too complex.

Reply:
I made the default for ordinal dependent on what categories are.